scheduler.go

```diff
@@ -275,7 +286,16 @@ func (s *Scheduler) run(job *Job) error {
 	}
 
 	job.lastRun = s.time.Now(s.loc)
-	go job.run()
+
+	if s.sem != nil {
+		s.sem.Acquire(context.Background(), 1)
```

Comment:
If the semaphore is acquired here and locked until freed, job.lastRun (line 288) will have an incorrect value. Maybe we should this check be at the start of the function?

Reply:
Good catch. Having it in the beginning also enables another scheduler to lock the job if it's waiting to acquire the semaphore too.

Reply:
@Streppel, what happens if the lock is already acquired by the time it attempts to lock now? Does the goroutine leak (blocked forever) and the semaphore is never released?

Reply:
The locker is meant for distributed computation. We'd actually need to put the semaphore after the locker check. Check my last comment at the thread